api/nodes/views.py

```diff
@@ -228,12 +294,62 @@ def get_default_odm_query(self):
         query = base_query & permission_query
         return query
 
-    # overrides ListCreateAPIView
+    # overrides ListBulkCreateUpdateDestroyAPIView
     def get_queryset(self):
-        query = self.get_query_from_request()
-        return Node.find(query)
+        # For bulk requests, queryset is formed from request body.
+        if isinstance(self.request.data, list):
+            query = Q('_id', 'in', [node['id'] for node in self.request.data])
+
+            user = self.request.user
+            if user.is_anonymous():
+                auth = Auth(None)
+            else:
+                auth = Auth(user)
 
-    # overrides ListCreateAPIView
+            nodes = Node.find(query)
+            for node in nodes:
+                if not node.can_edit(auth):
```

Comment:
I'd need to delve in more to offer a concrete suggestion; may take a look at the end of this pass. The alternatives may be worse, but I'm not yet 100% sold that duplicating permissions checks this way is the optimal choice.

The DRF docs seem to imply there's a way to turn on object-level permissions in a list view (emphasis added), but there may be other options.

> For performance reasons the generic views will not **automatically** apply object level permissions to each instance in a queryset when returning a list of objects.